event-sourcing/event-source-api/src/main/java/uk/gov/justice/services/eventsourcing/source/api/service/EventsService.java

```diff
@@ -0,0 +1,94 @@
+import uk.gov.justice.services.eventsourcing.repository.jdbc.Direction;
+import uk.gov.justice.services.eventsourcing.repository.jdbc.Position;
+import uk.gov.justice.services.eventsourcing.repository.jdbc.event.Event;
+import uk.gov.justice.services.eventsourcing.repository.jdbc.event.EventJdbcRepository;
+import uk.gov.justice.services.eventsourcing.source.api.feed.event.EventEntry;
+
+import java.sql.SQLException;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.UUID;
+import java.util.function.Function;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.json.JsonObject;
+
+@ApplicationScoped
+public class EventsService {
+
+    @Inject
+    private EventJdbcRepository repository;
+
+    public List<EventEntry> events(final Position position,
+                                   final Direction direction,
+                                   final long pageSize,
+                                   final UUID streamId) throws SQLException {
+
+        final List<Event> entities = entitiesPage(position, direction, pageSize, streamId);
+
+        final List<EventEntry> feedEntries = feedEntries(entities);
+
+        return feedEntries;
+    }
+
+    private List<Event> entitiesPage(final Position position,
+                                     final Direction direction,
+                                     final long pageSize,
+                                     final UUID streamId) throws SQLException {
+        if (position.isHead()) {
+            return repository.head(pageSize, streamId).collect(toList());
+        }
+
+        if (position.isFirst()) {
+            return repository.first(pageSize, streamId).collect(toList());
+        }
+
+        if (FORWARD.equals(direction)) {
+            return repository.previous(position.getSequenceId(), pageSize, streamId).collect(toList());
+        }
+
+        if (BACKWARD.equals(direction)) {
+            return repository.next(position.getSequenceId(), pageSize, streamId).collect(toList());
+        }
+
+        return null;
+    }
+
+    public boolean recordExists(final long offset, final UUID streamId) {
+        return repository.recordExists(offset, streamId);
+    }
+
+    private List<EventEntry> feedEntries(final List<Event> eventStreams) {
+        return eventStreams.stream()
+                .map(toFeedEntry())
+                .collect(toList());
+    }
+
+    private Function<Event, EventEntry> toFeedEntry() {
+        return event -> new EventEntry(
+                event.getId(),
+                event.getStreamId(),
+                event.getSequenceId(),
+                event.getName(),
+                convert(event.getPayload()),
+                convertToTimestamp(event.getCreatedAt())
+        );
+    }
+
+    private String convertToTimestamp(final ZonedDateTime createdAt) {
+        return ZonedDateTimes.toString(createdAt);
+    }
+
+    private JsonObject convert(final String payload) {
```

Comment:
convertToJsonObject if you want consistent naming with above. or payloadAsJsonObject